src/wasm/wasm-validator.cpp

```diff
@@ -3495,43 +3508,58 @@ void FunctionValidator::visitContBind(ContBind* curr) {
                "invalid second type in ContBind expression");
 }
 
-void FunctionValidator::visitContNew(ContNew* curr) {
+void FunctionValidator::visitSuspend(Suspend* curr) {
   // TODO implement actual type-checking
+  shouldBeTrue(!getModule() || getModule()->features.hasStackSwitching(),
+               curr,
+               "suspend requires stack-switching [--enable-stack-switching]");
+}
+
+void FunctionValidator::visitResume(Resume* curr) {
+  // TODO implement actual type-checking
+  shouldBeTrue(!getModule() || getModule()->features.hasStackSwitching(),
+               curr,
+               "resume requires stack-switching [--enable-stack-switching]");
+
   shouldBeTrue(
-    !getModule() || getModule()->features.hasTypedContinuations(),
+    curr->sentTypes.size() == curr->handlerBlocks.size(),
     curr,
-    "cont.new requires typed-continuatons [--enable-typed-continuations]");
+    "sentTypes cache in Resume instruction has not been initialized");
 
   shouldBeTrue((curr->contType.isContinuation() &&
                 curr->contType.getContinuation().type.isSignature()),
                curr,
-               "invalid type in ContNew expression");
+               "invalid type in Resume expression");
 }
 
-void FunctionValidator::visitResume(Resume* curr) {
+void FunctionValidator::visitResumeThrow(ResumeThrow* curr) {
   // TODO implement actual type-checking
   shouldBeTrue(
-    !getModule() || getModule()->features.hasTypedContinuations(),
+    !getModule() || getModule()->features.hasStackSwitching(),
     curr,
-    "resume requires typed-continuatons [--enable-typed-continuations]");
+    "resume_throw requires stack-switching [--enable-stack-switching]");
```

Comment:
Let's check that exceptions are enabled as well.

Reply:
Good point! Thanks.